x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/shrink_field.tsx

```diff
@@ -51,22 +63,50 @@ export const ShrinkField: FunctionComponent<Props> = ({ phase }) => {
       }}
       fullWidth
     >
-      <EuiFlexGroup>
-        <EuiFlexItem>
+      {isUsingShardSize === undefined ? null : (
+        <>
+          <EuiRadioGroup
+            options={[
+              {
+                id: `${phase}-configureShardCount`,
+                label: i18nTexts.editPolicy.shrinkCountLabel,
+                'data-test-subj': `${phase}-configureShardCount`,
+              },
+              {
+                id: `${phase}-configureShardSize`,
+                label: i18nTexts.editPolicy.shrinkSizeLabel,
+                'data-test-subj': `${phase}-configureShardSize`,
+              },
+            ]}
+            idSelected={
+              isUsingShardSize ? `${phase}-configureShardSize` : `${phase}-configureShardCount`
+            }
+            onChange={(id) => setIsUsingShardSize(id === `${phase}-configureShardSize`)}
+          />
+          <EuiSpacer />
           <UseField
             path={path}
```

Comment:
As you are swapping 2 fields in place and you don't want to share their state you need to add a `key`.

```js
<UseField
  path={path}
  key={path}
  ...
```